core/include/usBundle.h

```diff
@@ -47,143 +48,297 @@ typedef ServiceReference<void> ServiceReferenceU;
+     *
+     * The value of {@code STATE_UNINSTALLED} is 0x00000001.
+     */
+    STATE_UNINSTALLED = 0x00000001,
+
+    /**
+     * The bundle is installed but not yet resolved.
+     *
+     * A bundle is in the {@code STATE_INSTALLED} state when it has been installed in
+     * the Framework but is not or cannot be resolved.
+     *
+     * This state is visible if the bundle's code dependencies are not resolved.
+     * The Framework may attempt to resolve a {@code STATE_INSTALLED} bundle's code
+     * dependencies and move the bundle to the {@code STATE_RESOLVED} state.
+     *
+     * The value of {@code STATE_INSTALLED} is 0x00000002.
+     */
+    STATE_INSTALLED = 0x00000002,
+
+    /**
+     * The bundle is resolved and is able to be started.
+     *
+     * A bundle is in the {@code STATE_RESOLVED} state when the Framework has
+     * successfully resolved the bundle's code dependencies. These dependencies
+     * include:
+     * - None (this may change in future versions)
+     *
+     * Note that the bundle is not active yet. A bundle is put in the
+     * {@code STATE_RESOLVED} state before it can be started. The Framework may
+     * attempt to resolve a bundle at any time.
+     *
+     * The value of {@code STATE_RESOLVED} is 0x00000004.
+     */
+    STATE_RESOLVED = 0x00000004,
+
+    /**
+     * The bundle is in the process of starting.
+     *
+     * A bundle is in the {@code STATE_STARTING} state when its {@link #Start(int)
+     * Start} method is active. A bundle must be in this state when the bundle's
+     * {@link BundleActivator#Start(const BundleContext&)} is called.
+     * If the {@code BundleActivator#Start} method completes without exception, then
+     * the bundle has successfully started and moves to the {@code STATE_ACTIVE}
+     * state.
+     *
+     * If the bundle has a {@link Constants#ACTIVATION_LAZY lazy activation
+     * policy}, then the bundle may remain in this state for some time until the
+     * activation is triggered.
+     *
+     * The value of {@code STATE_STARTING} is 0x00000008.
+     */
+    STATE_STARTING = 0x00000008,
+
+    /**
+     * The bundle is in the process of stopping.
+     *
+     * A bundle is in the {@code STATE_STOPPING} state when its {@link #Stop(int)
+     * Stop} method is active. A bundle is in this state when the bundle's
+     * {@link BundleActivator#Stop(const BundleContext&)} method
+     * is called. When the {@code BundleActivator#Stop} method completes the bundle
+     * is stopped and moves to the {@code STATE_RESOLVED} state.
+     *
+     * The value of {@code STATE_STOPPING} is 0x00000010.
+     */
+    STATE_STOPPING = 0x00000010,
+
+    /**
+     * The bundle is now running.
+     *
+     * A bundle is in the {@code STATE_ACTIVE} state when it has been successfully
+     * started and activated.
+     *
+     * The value of {@code STATE_ACTIVE} is 0x00000020.
+     */
+    STATE_ACTIVE = 0x00000020
+  };
+
+  enum StartOptions : uint32_t {
```

Comment:
I think it would be a good idea to mark these options as not supported. 
Clients will most likely think there is full support for these options if they are documented.